azure-iot-hub-devicesdk/samples/receive_device_method.py

```diff
@@ -0,0 +1,84 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import os
+import json
+import logging
+import threading
+from six.moves import input
+from azure.iot.hub.devicesdk import DeviceClient
+from azure.iot.hub.devicesdk.auth.authentication_provider_factory import from_connection_string
+
+logging.basicConfig(level=logging.ERROR)
+
+
+# The connection string for a device should never be stored in code. For the sake of simplicity we're using an environment variable here.
+conn_str = os.getenv("IOTHUB_DEVICE_CONNECTION_STRING")
+# The "Authentication Provider" is the object in charge of creating authentication "tokens" for the device client.
+auth_provider = from_connection_string(conn_str)
+# For now, the SDK only supports MQTT as a protocol. the client object is used to interact with your Azure IoT hub.
+# It needs an Authentication Provider to secure the communication with the hub, using either tokens or x509 certificates
+device_client = DeviceClient.from_authentication_provider(auth_provider, "mqtt")
+
+
+# connect the client.
+device_client.connect()
+
+
+# define behavior for handling methods
+def method1_listener(device_client):
+    while True:
+        method_call = device_client.receive_method("method1")  # Wait for method1 calls
+        payload = json.dumps({"result": True, "data": "some data"})  # set response payload
+        status = 200  # set return status code
+        print("executed method1")
+        device_client.send_method_response(method_call, payload, status)  # send response
```

Comment:
>method_call [](start = 43, length = 11)

the more i see this the more i think this should be method_call.request_id rather than the full object.

Reply:
I'm not sure I can agree with that, but let's discuss after this PR.

---
In reply to: [267185229](https://github.com/Azure/azure-iot-sdk-python-preview/pull/61#discussion_r267185229) [](ancestors = 267185229)